File: Capture the Flag/src/main/java/me/yoursole/ctf/Commands/Stop.java
package me.yoursole.ctf.Commands;

import me.yoursole.ctf.DataFiles.GameData;
import me.yoursole.ctf.DataFiles.GameLoop;
import me.yoursole.ctf.DataFiles.Items.Flag;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import java.util.UUID;


public class Stop implements CommandExecutor {
        @Override
        public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
            if(GameData.it!=null){
                if (sender.isOp()) {
                    for (Player player : Bukkit.getOnlinePlayers()) {
                        player.getInventory().remove(Flag.flag);
                        PotionEffect hasteEffect = GameData.it.getPotionEffect(PotionEffectType.FAST_DIGGING);
                        player.setGlowing(false);
                    }
                    GameData.it.setGlowing(false);
                    GameData.scores.compute(GameData.it.getUniqueId(), (player, score) -> (score == null ? 0 : score) + 60);
                    GameData.it = null;
                    sender.sendMessage("Game Stopped");


                    UUID a = (UUID) GameData.scores.keySet().toArray()[0];
                    for(UUID uuid : GameData.scores.keySet()){
                        if(GameData.scores.get(uuid)>GameData.scores.get(a)){
                            a=uuid;
                        }
                    }
                    for(Player player : Bukkit.getOnlinePlayers()){
                        player.sendMessage(ChatColor.GREEN+"The winner is "+ Bukkit.getPlayer(a).getDisplayName()+"!");
                    }
                    GameData.scores.clear();
                } else {
                    sender.sendMessage("don't u dare ._.");
                }
            }else{
                sender.sendMessage(ChatColor.RED+"The game is not running");
            }

            return true;

        }

    }

File: src/main/kotlin/me/yoursole/ctf/commands/Stop.kt
package me.yoursole.ctf.commands

import com.mojang.brigadier.CommandDispatcher
import me.yoursole.ctf.datafiles.GameData
import me.yoursole.ctf.datafiles.WorldManager
import me.yoursole.ctf.datafiles.items.Flag
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands.literal
import net.minecraft.network.chat.TextComponent
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import java.util.*

object Stop : BrigadierCommand {
    override fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(literal("stop").requiresOp().executes { ctx ->
            if (!GameData.gameRunning) {
                ctx.source.sendFailure(TextComponent("Game is not running!"))
                return@executes 0
            }
            for (player in Bukkit.getOnlinePlayers()) {
                player.inventory.remove(Flag.flag)
                player.isGlowing = false
            }

            if (Bukkit.getScheduler().isQueued(Start.id))
                Bukkit.getScheduler().cancelTask(Start.id)

            WorldManager.deleteWorlds()
            if (GameData.it != null) {
                GameData.it!!.isGlowing = false
                GameData.scores.compute(GameData.it!!.uniqueId) { _: UUID?, score: Int? -> (score ?: 0) + 60 }
            }
            GameData.it = null
            GameData.droppingPos = null
            GameData.dropLoc = null
            ctx.source.sendSuccess(TextComponent("Game Stopped!"), true)
            val a = GameData.scores.entries.maxByOrNull { it.value }?.key
            if (a != null) {
                for (player in Bukkit.getOnlinePlayers()) {
                    player.sendMessage("§aThe winner is ${Bukkit.getPlayer(a)?.displayName}!")
                }
            }
            GameData.scores.clear()
            GameData.gameRunning = false
            1
        })
    }
}